crates/graphql-composition/src/compose/directives.rs

```diff
@@ -86,7 +86,20 @@ pub(super) fn collect_composed_directives<'a>(
                         })
                     }
                     (None, true) => Some(ir::DirectiveProvenance::ComposeDirective),
-                    (None, false) => None,
+                    (None, false) => {
+                        if ctx.strict {
+                            ctx.diagnostics.push_fatal(format!(
```

Comment:
At this point in the composition process, I think the directives you don't want to be ignored are already filtered out. Have you tried doing this in `ingest_subgraphs::directives.rs` in the NoMatch branch? (there's already a warning being pushed there)